### ISSUE TITLE
Some of "is string" expressions should be substituted with "is StringValue"

### ISSUE THREAD

**Issue by tomekpiotrowski** [KSP-KOS/KOS#1357]
After the introduction of encapsulation types some of `is string` expressions should be substituted by `is StringValue`. Currently, for example, `processor(stringTag)` will fail because `stringTag is string` is false.

We should investigate whether other types can cause problems somewhere as well (`is int`, `is double` etc).

**Comment by hvacengi** [comment]
Related to #1033.  In fact, the use of the new encapsulation types (and the fixing of this bug) may actually significantly help with the boxing issues.

**Comment by tomekpiotrowski** [comment]
This was automatically closed when #1358 got merged, but I think it's worth reopening since there are potentially other similar bugs (related to `is int` and others) that are not fixed yet.

**Comment by hvacengi** [comment]
I just went through all of the functions, and they don't appear to have any casts that we need to be worried about. The opcodes will be much more difficult to manually walk, because function pointers should still be int types.  I only found a single cast to double in `OpcodeMathNegate` which is properly guarded and falls back to check for an `op_UnaryNegation` method using reflection.  I'm not sure that we for sure have *everything* but it looks to me like we should have almost everything.  

The best thing we can do now is test as many scripts as they can before release.  I'm going to pull out all of my old scripts and test.
